### PR TITLE
routing: move payment session to separate file

### PR DESCRIPTION
**Pull request by joostjager**
Preparation to keep better overview for upcoming routing related work.

### CODE REVIEW

File: routing/router.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/coreos/bbolt"
 	"github.com/davecgh/go-spew/spew"
 	"github.com/go-errors/errors"
-	"github.com/lightningnetwork/lightning-onion"
+	sphinx "github.com/lightningnetwork/lightning-onion"
```
Comment (halseth) [inline routing/router.go]: unrelated change? :p

Comment (joostjager) [reply routing/router.go]: It's `goimports` again.

**Comment by joostjager** [comment]
@halseth rebased, review dismissed, ptal

File: routing/payment_session.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+func (p *paymentSession) ReportEdgeFailure(e *edgeLocator) {
+	log.Debugf("Reporting edge %v failure to Mission Control", e)
+
+	// First, we'll add the failed edge to our local prune view snapshot.
+	p.pruneViewSnapshot.edges[*e] = struct{}{}
+
+	// With the edge added, we'll now report back to the global prune view,
+	// with this new piece of information so it can be utilized for new
+	// payment sessions.
+	p.mc.Lock()
```
Comment (cfromknecht) [inline routing/payment_session.go]: Not related to this PR, tho we should abstract this operation behind a method to prevent us from modifying mission control’s internal state directly and externally acquiring its mutexes. Same for the above method

Better yet, mission control itself deserves its own interface so that access to internal variables is prevented entirely

Comment (joostjager) [reply routing/payment_session.go]: Yes, the design of mission control definitely needs to be stepped up.
